contentcuration/contentcuration/viewsets/base.py

```diff
@@ -967,12 +1024,16 @@ def update_progress(progress=None):
                     'task_id': task_id,
                     'task_name': task_name,
                     'traceback': traceback_str,
-                    'progress': progress,
+                    'progress': custom_task_metadata_object.progress,
                     'channel_id': channel_id,
                     'status': status,
                 }
             }
         )
+        task_object.status = status
+        task_object.traceback = traceback_str
+        task_object.save()
+        raise
     finally:
         if status == states.STARTED:
```

Comment:
Ahh this should be task_object.status now! I think this was the reason for `test_sync_channel_called_correctly` failing.

Reply:
The test failures were a result of trying to save the objects during the change event application - so instead I have followed the lead of your code, and just removed that in favour of transmitting the progress tracking via the websockets instead.